### ISSUE TITLE
Merge by distance Node, must permit Vertex only input.

### ISSUE THREAD

**Issue by zeffii** [nortikin/sverchok#4037]
https://github.com/nortikin/sverchok/blob/master/nodes/modifier_change/merge_by_distance.py

![image](https://user-images.githubusercontent.com/619340/114270862-e52bba80-9a0e-11eb-88e2-734124a5c92c.png)

snlite script
```python
"""
in vfields v
in doubles_distance s d=0.0 n=2
out verts v
"""
import bmesh

for vfield in vfields:
    if doubles_distance > 0.0:
        bm = bmesh_from_pydata(vfield, [], [])
        bmesh.ops.remove_doubles(bm, verts=bm.verts[:], dist=doubles_distance)

        vfield_mod, _, _ = pydata_from_bmesh(bm)
        verts.append(vfield_mod)
    else:
        verts.append(vfield)


```

**Comment by zeffii** [comment]
@vicdoval 

```diff
        poly_edge = self.inputs['PolyEdge']
-        poly_edge.nesting_level = 3
+        poly_edge.nesting_level = 2
        poly_edge.default_mode = 'EMPTY_LIST'
```
?

**Comment by zeffii** [comment]
![GIF 4-10-2021 3-55-15 PM](https://user-images.githubusercontent.com/619340/114272150-3c348e00-9a15-11eb-8326-eb2a576fc5e5.gif)

**Comment by zeffii** [comment]
i won't push until you have a look if that change has any knock-on effects which i'm oblivious to.

**Comment by vicdoval** [comment]
Yes, well... the nesting level of pols and edges is always 3 so there is a ensure_min_nesting function in the recursive nodes mixin to enable also working with data with less nesting as [[0,1,2]].. 
So my solution was  a general fix to that  in #4033 https://github.com/nortikin/sverchok/blob/344ab0b084f4af99c35bc342f2e1a411539df8e8/utils/sv_bmesh_utils.py#L85

https://github.com/nortikin/sverchok/blob/356e6dbb97cf4ede0bbb48e75ead00bf1b7502db/data_structure.py#L1078-L1083

**Comment by zeffii** [comment]
so I'll close my PR then
